vars/publishBuildStatusReport.groovy

```diff
@@ -0,0 +1,42 @@
+/**
+ * Generates and publishes a JSON build status report for monitoring Jenkins builds.
+ *
+ * This function deploys and executes a shell script that handles all processing:
+ * hostname extraction, path construction, JSON generation, and upload to Azure storage.
+ *
+ * IMPORTANT: Only runs on principal branches (BRANCH_IS_PRIMARY=true).
+ *
+ * Example usage:
+ *
+ * @Library('pipeline-library@main') _
+ * buildDockerAndPublishImage('404', [
+ *     targetplatforms: 'linux/amd64,linux/arm64',
+ * ])
+ * node('linux-arm64') {
+ *     stage('[Post-Build] Status Report') {
+ *         publishBuildStatusReport()
+ *     }
+ * }
+ */
+def call(Map config = [:]) {
+  // if (env.BRANCH_IS_PRIMARY == null || !env.BRANCH_IS_PRIMARY.toBoolean()) {
```

Comment:
```suggestion
  // if (!env.BRANCH_IS_PRIMARY) {
```

Ref. https://docs.groovy-lang.org/latest/html/documentation/core-semantics.html#the-groovy-truth and https://docs.groovy-lang.org/latest/html/documentation/core-semantics.html#_object_references in particular

=> Groovy considers `null` to be `false` when called in a logical expression.